src/client.rs

```diff
@@ -461,7 +480,9 @@ where
             Some(cache) => cache,
         };
         let enabled = cache.is_enabled_str(feature_name, Some(context), false, &self.cached_state);
+        let feature = &cache.str_features.get(feature_name);
```

Comment:
As mentioned in the PR description, most of it is more or less the same. The one instance that does see significant changes is parallel unknown-features(str). When you try and get a feature that doesn't exist, we initialize a new dashmap for the feature now. I suspect that this line is what causes that.

Reply:
I'm a banana, thanks, I didn't read the updated description. 24% is significant but that should be cold path so I'm not that worried. Everything else looks like statistical noise to me

Reply:
> 24% is significant but that should be cold path

Yeah, this is my thinking as well. I don't imagine you'd check _new_ unknown features in a loop.

> Everything else looks like statistical noise to me

Agreed. So nothing in most cases, but a marked regression for what should be a very cold path.

I did some digging and realized that the initialization of the dashmap happens when you check `is_enabled_str` with an unknown feature. That happens regardless of whether this line was moved earlier or not. 

I've got an idea that might work, however: if we store unknown variant stats on the cache itself, then we'll save some initialization work. That might work. The tradeoff is that you'll always carry that thing around. I'll see what I can do.

Reply:
Thinking about this a bit more, I don't think that idea will work either.

The core of this is around [this clause in `is_enabled_str`](https://github.com/Unleash/unleash-client-rust/blob/2419b778eba71ea038958f9267b2c1de6da27c1b/src/client.rs#L382-L436).

What happens is that if the feature is unknown, we currently:
- create a new cached state from the old state, cloning old features and adding them
- creating a new feature for the unknown feature and inserting that into the cache
- replacing the old cached state with the new state

My thought was that if the unknown str feature doesn't need to initialize its own map, then we could save some there. However, if unknown str features could **not** have variant counts, then they have to be different from other features. This would likely cause a larger restructuring.

Maybe (_maybe_!) we could avoid storing unknown str features as actual features and just store them in a map (I think they're only created for metrics' sake), but that would also be a lot of restructuring work and I don't think it's worth it (at least not for now).

In short, I think the regressions for the cold path here are acceptable and vote we go forward with it.

Let me know if you disagree.